frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx

```diff
@@ -1,144 +1,52 @@
-  }
-
-  const secretResource = secret.data;
-
-  if (_.isEmpty(secretResource)) return <ErrorPage404 />;
-
-  const secretName = secretResource[0]?.metadata.name;
-  const releaseName = helmReleaseData?.name;
-
-  const title = (
-    <>
-      {releaseName}
-      <Badge isRead style={{ verticalAlign: 'middle', marginLeft: 'var(--pf-global--spacer--md)' }}>
-        <Status status={_.capitalize(helmReleaseData?.info?.status)} />
-      </Badge>
-    </>
-  );
-
-  const menuActions = [
-    () => upgradeHelmRelease(releaseName, namespace),
-    () => deleteHelmRelease(releaseName, namespace, `/helm-releases/ns/${namespace}`),
-  ];
-
-  return (
-    <DetailsPage
-      kindObj={SecretModel}
-      match={match}
-      menuActions={menuActions}
-      name={secretName}
-      namespace={namespace}
-      customData={helmReleaseData}
-      breadcrumbsFor={() => [
-        {
-          name: `Helm Releases`,
-          path: `/helm-releases/ns/${namespace}`,
-        },
-        { name: `Helm Release Details`, path: `${match.url}` },
-      ]}
-      title={title}
-      kind={SecretReference}
-      pages={[
-        navFactory.details(HelmReleaseOverview),
-        {
-          href: 'resources',
-          name: 'Resources',
-          component: HelmReleaseResources,
-        },
-        {
-          href: 'history',
-          name: 'History',
-          component: HelmReleaseHistory,
-        },
-        {
-          href: 'releasenotes',
-          name: 'Release Notes',
-          component: HelmReleaseNotes,
-        },
-      ]}
-      customKind={HelmReleaseReference}
-    />
-  );
 };
 
-const HelmReleaseDetailsPage: React.FC<HelmReleaseDetailsPageProps> = ({ secret, match }) => {
+const HelmReleaseDetailsPage: React.FC<HelmReleaseDetailsProps> = ({ match }) => {
   const namespace = match.params.ns;
   const helmReleaseName = match.params.name;
-
-  const [helmReleaseData, setHelmReleaseData] = React.useState<any>();
-
-  React.useEffect(() => {
-    let ignore = false;
-
-    const getHelmReleases = async () => {
-      try {
-        const helmReleases = await fetchHelmReleases(namespace);
-        if (!ignore) {
-          const releaseData = helmReleases.find((release) => release.name === helmReleaseName);
-          setHelmReleaseData(releaseData);
-        }
-        // eslint-disable-next-line no-empty
-      } catch {}
-    };
-
-    getHelmReleases();
-
-    return () => {
-      ignore = true;
-    };
-  }, [helmReleaseName, namespace]);
-
   return (
-    <LoadedHelmReleaseDetailsPage match={match} secret={secret} helmReleaseData={helmReleaseData} />
+    <>
+      <NamespacedPage
+        variant={NamespacedPageVariants.light}
+        hideApplications
+        onNamespaceChange={handleNamespaceChange}
+      >
+        <Firehose
+          resources={[
+            {
+              namespace,
+              kind: SecretModel.kind,
+              prop: 'secret',
```

Comment:
can use `SecretModel.id` here

Reply:
To be fair, this is very specific to the individual use of the result data.
Using an arbitrary string is perfectly fine here.